src/tribler/core/components/restapi/restapi_component.py

```diff
@@ -115,10 +120,11 @@ async def run(self):
                        metadata_store_component.mds)
         self.maybe_add('/tags', TagsEndpoint, db=tag_component.tags_db, community=tag_component.community)
 
-        ipv8_root_endpoint = IPV8RootEndpoint()
-        for _, endpoint in ipv8_root_endpoint.endpoints.items():
-            endpoint.initialize(ipv8_component.ipv8)
-        self.root_endpoint.add_endpoint('/ipv8', ipv8_root_endpoint)
+        if not isinstance(ipv8_component, NoneComponent):
```

Comment:
If I understand correctly, with this logic `/ipv8` endpoint with its children will not be included in the generated API doc. How about including `IPV8RootEndpoint` unconditionally and then skipping `endpoint.initialize(...)` calls to all children endpoints in swagger mode?